compiler-rt/lib/sanitizer_common/sanitizer_mac.h

```diff
@@ -37,16 +37,18 @@ struct VersionBase {
 
   VersionBase(u16 major, u16 minor) : major(major), minor(minor) {}
 
-  bool operator==(const VersionType &other) const {
-    return major == other.major && minor == other.minor;
-  }
   bool operator>=(const VersionType &other) const {
     return major > other.major ||
            (major == other.major && minor >= other.minor);
   }
   bool operator<(const VersionType &other) const { return !(*this >= other); }
 };
 
+template <typename VersionType>
```

Comment:
Will this work?

```
struct VersionBase {
  u16 major;
  u16 minor;

  VersionBase(u16 major, u16 minor) : major(major), minor(minor) {}

  template <typename VersionType>
  bool operator==(const VersionType &other) const {
    return major == other.major && minor == other.minor;
  }

  template <typename VersionType>
  bool operator>=(const VersionType &other) const {
    return major > other.major ||
           (major == other.major && minor >= other.minor);
  }

  template <typename VersionType>
  bool operator<(const VersionType &other) const { return !(*this >= other); }
};
```

Reply:
I guess this will defy the purpose of two different types.

then probably:
```
template <typename VersionType>
bool operator==(const VersionBase<VersionType> &self, const VersionBase<VersionType> &other) {
  return self.major == other.major && self.minor == other.minor;
}
```

Reply:
It doesn't.  The code (e.g. MacosVersion immediately below) expects VersionBase to be a template.  Defining an matching `operator!=` seems to work.

Reply:
but it's the same issue, we declare operator which can be a candidate for any type

Reply:
I see you replied to first one, which is wrong.

However this one should work. I'll try locally
```
template <typename VersionType>
bool operator==(const VersionBase<VersionType> &self, const VersionBase<VersionType> &other) {
  return self.major == other.major && self.minor == other.minor;
}
```